src/components/TreebankService/TreebankService.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import {
+  MessagingService,
+  ResponseMessage,
+  WindowIframeDestination as Destination,
+} from 'alpheios-messaging/dist/dev/alpheios-messaging';
+
+const config = {
+  name: 'treebank',
+  targetIframeID: 'string-not-used',
+  targetURL: 'string-not-used',
+};
+
+const error = (request, message) => ResponseMessage.Error(request, new Error(message));
+
+class TreebankService extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { redirectTo: null };
+
+    this.service = new MessagingService('treebank-service', new Destination(config));
+
+    this.messageHandler = this.messageHandler.bind(this);
+  }
+
+  componentDidMount() {
+    const { name } = config;
+
+    // TODO this line is causing the error (even with try { }):
+    // TypeError: Cannot set property 'origin' of undefined
+    this.service.registerReceiverCallback(name, this.messageHandler);
+  }
+
+  componentWillUnmount() {
+    // TODO is there any way to unregister in alpheios-messaging?
+    // this.service.unregisterReceiverCallback(name, this.messageHandler);
+  }
+
+  messageHandler(request, responseFn) {
+    const { body } = request;
+    const [name] = Object.keys(body);
+
+    switch (name) {
+      case 'gotoSentence':
+        this.setState({ redirectTo: body[name].sentenceId });
+
+        // TODO what should the message be? Should it be an object or a string?
+        responseFn(ResponseMessage.Success(request, 'success'));
```

Comment:
I wasn't sure what to respond with. In https://github.com/perseids-publications/treebank-template/pull/15 it seems like the response is always `{}`.

Reply:
I wasn't sure what do here either. Maybe an object with status?  E.g. `{ status: 'success' }`

Reply:
That makes sense to me.